tests/test_salcommander_minimal_controllers.py

```diff
@@ -0,0 +1,145 @@
+# This file is part of ts_SalMultiLanguageTests.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import subprocess
+import pathlib
+import unittest
+import warnings
+
+from lsst.ts import salobj
+
+# Long enough to perform any reasonable operation
+# including starting a CSC or loading a script (seconds)
+STD_TIMEOUT = 60
+
+INITIAL_LOG_LEVEL = 20
+
+index_gen = salobj.index_generator()
+
+languages = ["cpp", "java"]
+
+
+class SalCommanderToMinimalControllerTestCase(unittest.TestCase):
+    """Test a C++ or Java Commander communicating to
+    various minimal controllers."""
+
+    def setUp(self) -> None:
+        salobj.set_random_lsst_dds_partition_prefix()
+        self.index = next(index_gen)
+
+    def getLanguageVariables(self, language) -> None:
```

Comment:
Please annotate the type of the `language` argument. In this case str, though I suggest an enum below.

Also please consider renaming to `get_language_variables` to match T&S method naming convention. (I know setUp is an outlier, but that is forced by `unittest`, which is an older library).

Reply:
Fixed.